library/src/G4CMPSarkisNIEL.cc

```diff
@@ -56,25 +50,31 @@ PartitionNIEL(G4double energy, const G4Material *material,G4double /*Zin = 0.*/,
   // Check if the material is silicon or similar (within +-1 of Z and A of silicon)
     
   if (std::abs(Z - SiZ) <= 1.0 && std::abs(A - SiA) <= 1.0) {
+    if (firstCall){
       G4Exception("G4CMPImpactTunlNIEL", "G4CMP1005", JustWarning,
                   "Sarkis model is obtained in the range of 50 eV to 3 MeV. Above 3 MeV, Lindhard model will be used.");
-    
+      firstCall = false;
+    }
     // Sarkis model below 3 MeV 
     if (energy <= 3 * MeV) {
-        // Sarkis model function
-        
-        inputFile.open(fPath);                       // open the data file
-        lVector.Retrieve(inputFile, false);          // load the data from the text file
-        inputFile.close();
-        return lVector.Value(energy, idx);           // do the interpolation and return the yield value
+      // Sarkis model function
+      std::ifstream inputFile;
+      inputFile.open(fPath);                       // open the data file
+      lVector.Retrieve(inputFile, false);          // load the data from the text file
+      inputFile.close();                           // close the data file 
```

Comment:
You can leave this out if you like (or not).  Since the input stream is now scoped, it will be deleted (and the file closed) automatically at `return`.